Doppler.Sap/Controllers/BusinessPartnerController.cs

```diff
@@ -0,0 +1,71 @@
+using Doppler.Sap.Models;
+using Doppler.Sap.Services;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+using System;
+using System.Threading.Tasks;
+
+namespace Doppler.Sap.Controllers
+{
+    [Route("[controller]")]
+    [ApiController]
+    [Authorize]
+    public class BusinessPartnerController
+    {
+        private readonly ILogger<BillingController> _logger;
+        private readonly IBusinessPartnerService _businessPartnerService;
+
+        public BusinessPartnerController(ILogger<BillingController> logger, IBusinessPartnerService billingService) =>
+            (_logger, _businessPartnerService) = (logger, billingService);
+
+        [HttpPost("CreateOrUpdateBusinessPartner")]
+        public async Task<IActionResult> CreateOrUpdateBusinessPartner([FromBody] DopplerUserDto dopplerUser)
+        {
+            _logger.LogInformation($"Received user: {dopplerUser.Email}");
+            var userVerificationError = VerifyUserInformation(dopplerUser);
+            if (!string.IsNullOrEmpty(userVerificationError))
+            {
+                return new BadRequestObjectResult(userVerificationError);
+            }
+
+            try
+            {
+                await _businessPartnerService.CreateOrUpdateBusinessPartner(dopplerUser);
+
+                return new OkResult();
+            }
+            catch (Exception e)
+            {
+                _logger.LogError($"Failed at mapping data from user: {dopplerUser.Id}, Object sent: {JsonConvert.SerializeObject(dopplerUser)} ", e);
```

Comment:
Minor change. I think the scope of this catch is greater now. Maybe "Failed at creating/updating user" would be more accurate. What do you think?

Reply:
What do you think uses the log.error with the Exception passed as a parameter?
`_logger.LogError(e, $"Failed at mapping data from user: {dopplerUser.Id}, Object sent: {JsonConvert.SerializeObject(dopplerUser)} ");`

Reply:
@nreal The exception is at the end, as parameter. Is this what do you say?

Reply:
> Minor change. I think the scope of this catch is greater now. Maybe "Failed at creating/updating user" would be more accurate. What do you think?

You're right @fgchaio , now there isnt a mapping in BusinessPartnerService anymore.

Reply:
https://github.com/FromDoppler/doppler-sap/pull/20#discussion_r510368843
Yes, you are passed as parameter the exception but is not used in the string message, if you use the variable `e` in the first parameter, you are using the exception log correctly, see attachment
![image](https://user-images.githubusercontent.com/6796523/96914813-dad72d80-147b-11eb-8be4-20d5a60ebe42.png)

Reply:
Ohhh you're right, I wasnt pay attention about that! I will change it.